src/components/ContentWidgets/ToDoList/ToDoListEditor.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { WidgetEditorProps } from '../../ContentMapping/ContentMapping';
+import { ToDoListItem } from './ToDoList';
+
+export const ToDoListEditor: React.FC<WidgetEditorProps> = ({
+    originalContent,
+    editedContent,
+    setEditedContentOnChange
+}) => {
+    let listContent = null;
+
+    // let findItemIndex = (findItemID: number, copyEditedContent: ToDoListItem[]) => {
+        
+    //     for (let i = 0; i < copyEditedContent.length; i++) {
+    //         if (findItemID == copyEditedContent[i].toDoListItem_order) {
+    //             return i;
+    //         }
+    //     }
+    //     return -1;
+    // }
+
+    if (editedContent.toDoList_content != null) {
+        listContent = editedContent.toDoList_content.map((item) => {
+            
+            let copyEditedContent = [...editedContent.toDoList_content!!];
+
+            let removeAction = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+                let index = parseInt(e.currentTarget.parentElement!!.id);
+                copyEditedContent.splice(index, 1);
+                // reassigns ordering
+                for (let i = 0; i < copyEditedContent.length; i++) {
+                    copyEditedContent[i].toDoListItem_order = i;
+                }
+                setEditedContentOnChange("toDoList_content", copyEditedContent);
+            }
+
+            let changeInput = (e: React.FormEvent<HTMLInputElement>) => {
+                let index = parseInt(e.currentTarget.parentElement!!.id);
+                // console.log(parseInt(e.currentTarget.parentElement!!.id.charAt(e.currentTarget.parentElement!!.id.length - 1)));
+                copyEditedContent[index].toDoListItem_content = e.currentTarget.value;
+                setEditedContentOnChange("toDoList_content", copyEditedContent);
+            }
+
+            let changeCompleted = (e: React.FormEvent<HTMLInputElement>) => {
+                let index = parseInt(e.currentTarget.parentElement!!.id);
```

Comment:
This is repeated multiple times in these functions and that should tell you it should be factored out in some way, and like I mentioned in a previous comment it would be best to have these functions take in the index as a parameter and instead do something like `onClick={(e) => changeCompleted(e, index)}`